### DISCUSSION TITLE
Podman Mac M1 use gdb through QEMU

### DISCUSSION THREAD

**Discussion by glli01** [containers/podman#21197] [Q&A]
Hi, I'm on an M1 Mac (arm/aarch64) running a machine (Oracle Linux 8) through QEMU (--platform linux/amd64).

I've been unable to run a command (due to a segmentation fault), and have been trying to get gdb or lldb to work.
Does anyone know how to set this up? The issue is that ptrace isn't enabled on qemu, so the solution would be to use their gdb remote server from: https://qemu-project.gitlab.io/qemu/system/gdb.html.

But for my (basic) understanding of podman, the podman machine runs qemu in the background - so how would I add the necessary -s and -S flags/ open up ports for gdb/lldb from the above link?

Error when running GDB:
`(gdb) run
Starting program: /usr/bin/zsh 
warning: Could not trace the inferior process.
Error: 
warning: ptrace: Function not implemented
During startup program exited with code 127.
(gdb) `

**Comment by rhatdan** [comment]
Have you tried to ssh into the VM and running the podman command in linux?

podman machine ssh
